### ISSUE TITLE
TemplateSpectralModel fails for single energy bin

### ISSUE THREAD

**Issue by registerrier** [gammapy/gammapy#3881]
**Gammapy version**
dev

**Bug description**
`TemplateSpectralModel` fails on init when the input arrays have a single elements. The error raised is:
```
ValueError: Interpolating scalar values requires using method='nearest' explicitely.
```
This is likely connected with the change introduced in PR #3837 

Note that this is problematic because `likelihood` type light curves cannot be serialized from disk.

**Expected behavior**
The model initialization should proceed.

**To Reproduce**
```
import astropy.units as u
from gammapy.modeling.models import TemplateSpectralModel
TemplateSpectralModel([10]*u.TeV, [1])
```

**Comment by registerrier** [comment]
The test pass because it uses the `sed_type=flux`. 

The model `init` works in v0.19 but fails on evaluate.

**Comment by registerrier** [comment]
This is solved by #3882. Closing
